implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,362 @@
+                    }
+                    return null;
+                })
+                .filter(Objects::nonNull);
+    }
+
+    private Metric.Builder createMetricBuilder(Meter meter, String metricKey) {
+        return metricBuilderFactory
+                .newMetricBuilder(metricKey)
+                .setDimensions(fromTags(meter.getId().getTags()))
+                .setTimestamp(Instant.ofEpochMilli(clock.wallTime()));
+    }
+
+    private String createMetricKey(String name, String tagValueRepresentation) {
+        if (name.endsWith(".percentile")) {
+            return null;
+        }
+
+        String dynatraceTag;
+        switch (tagValueRepresentation) {
+            case "total":
+                dynatraceTag = "sum";
+                break;
+            case "value":
+                dynatraceTag = "";
+                break;
+            case "percentile":
+                // drop lines that have a tag value representation of percentile.
+                // we use the 0 percentile to determine minimum values for summaries and timers
+                // so this will export 0 for every timer and summary.
+                logger.warn("dropping percentile value");
+                return null;
+            default:
+                dynatraceTag = tagValueRepresentation;
+                break;
+        }
+
+        if (dynatraceTag.isEmpty()) {
+            return name;
+        }
+        return String.format("%s.%s", name, dynatraceTag);
+    }
+
+    private DimensionList fromTags(List<Tag> tags) {
+        return DimensionList.fromCollection(
+                tags.stream()
+                        .map(x -> Dimension.create(x.getKey(), x.getValue()))
+                        .collect(Collectors.toList()));
+    }
+
+    static <T> Stream<T> streamOf(Iterable<T> iterable) {
+        return StreamSupport.stream(iterable.spliterator(), false);
+    }
+
+    private void send(List<String> metricLines) {
+        try {
+            String body = metricLines.stream().collect(Collectors.joining(System.lineSeparator()));
+            logger.debug("sending lines:\n" + body);
+
+            httpClient.post(endpoint)
+                    .withHeader("Authorization", "Api-Token " + config.apiToken())
+                    .withPlainText(body)
+                    .send()
+                    .onSuccess((r) -> logger.info("Ingested {} metric lines into Dynatrace, response: {}", metricLines.size(), r.body()))
+                    .onError((r) -> logger.error("Failed metric ingestion. code={} body={}", r.code(), r.body()));
+        } catch (Throwable throwable) {
+            logger.error("Failed metric ingestion: {}", throwable.getMessage());
+        }
+    }
+
+    private void sendInBatches(List<String> metricLines) {
+        MetricLinePartition.partition(metricLines, config.batchSize())
```

Comment:
The batch size must not exceed 1000 lines as per the API docs. This should be documented in the Micrometer config.